### PR TITLE
Project Setup

### LINKED ISSUE #1: Set up TypeScript repository
The canonical issue 1!

We want this repo to have:

1. TypeScript support / transpiling
2. Linting + editorconfig
3. Tests
4. GitHub CI
5. A README + Code of Conduct

### PR DESCRIPTION
**Pull request by slifty**
This PR sets up the repository with our standard dev tooling.

This includes linting, testing, transpiling, type checking, and GitHub CI.}

Most of the content is pulled straight from [upload-service](https://github.com/PermanentOrg/upload-service).  Changes include:

1. Adding linting to our CI
2. Using Node 16
3. Some minor style fixes to settings files
4. Tests pass even if there are no tests written

Resolves #1

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 2
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
```
Comment (cecilia-donnelly) [inline .editorconfig:9]: Thanks for including this!

File: package.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "node": ">=16.0"
+  },
+  "private": true,
+  "devDependencies": {
+    "@babel/cli": "^7.16.7",
+    "@babel/core": "^7.16.7",
+    "@babel/node": "^7.16.7",
+    "@babel/preset-env": "^7.16.7",
+    "@babel/preset-typescript": "^7.16.7",
+    "@tsconfig/node12": "^1.0.9",
```
Comment (cecilia-donnelly) [inline package.json:40]: Can we use node16 here and in the tsconfig?

**Review by cecilia-donnelly — CHANGES_REQUESTED** [review]
All seems very reasonable!  Thanks for getting this put together, @slifty .  I think you switched node versions partway through working on this, so we want [the config for 16](https://www.npmjs.com/package/@tsconfig/node16) I believe.

File: package.json (hunk 2 above)
Comment (slifty) [reply package.json:40]: My goodness this shows up in so many places -- great catch thank you.

**Review by cecilia-donnelly — APPROVED** [review]
:tada:

**Pull request merged**